File: CODIGOS_PYTHON/ACT6.py
class ObjetoCompuesto:
    def __init__(self, valor):
        self.atributo = valor

    def get_atributo(self):
        return self.atributo

    def set_atributo(self, valor):
        self.atributo = valor


class Base:
    def __init__(self, compuesto):
        self.objeto_compuesto = compuesto

    def get_objeto_compuesto(self):
        return self.objeto_compuesto

    def modificar_atributo(self, valor):
        self.objeto_compuesto.set_atributo(valor)

    def eliminar_objeto(self):
        del self.objeto_compuesto


if __name__ == "__main__":
    objeto = ObjetoCompuesto(10)
    base = Base(objeto)

    compuesto = base.get_objeto_compuesto()
    compuesto.set_atributo(20) #Valor del atributo establecido en 20
    print("Atributo del objeto:", compuesto.get_atributo())

    base.modificar_atributo(30) #Modificando el atributo del objeto
    print("Atributo del objeto modificado:", compuesto.get_atributo())

    base.eliminar_objeto() #Para eliminar el objeto

File: CODIGOS_JAVA/ACT6.java
class ObjetoCompuesto {  // Clase del objeto que se compone
    private int atributo;
    public ObjetoCompuesto(int valor) {  // Constructor con un parámetro
        atributo = valor;
    }
    public int getAtributo() {  // Método para obtener el atributo
        return atributo;
    }
    public void setAtributo(int valor) {  // Método para establecer el atributo
        atributo = valor;
    }
}
class Base {  // Clase base que representa la entidad principal en la relación de composición
    private ObjetoCompuesto objetoCompuesto;  // Atributo que representa el objeto compuesto
    public Base(ObjetoCompuesto compuesto) {  // Constructor con un objeto de la clase ObjetoCompuesto
        objetoCompuesto = compuesto;
    }
    public ObjetoCompuesto getObjetoCompuesto() {  // Método para acceder al objeto compuesto
        return objetoCompuesto;
    }
    public void modificarAtributo(int valor) {  // Método para modificar el atributo del objeto compuesto
        objetoCompuesto.setAtributo(valor);
    }
    public void eliminarObjeto() {  // Método para eliminar el objeto compuesto
        objetoCompuesto = null;
    }
}
public class Act6 {
    public static void main(String[] args) {
        ObjetoCompuesto objeto = new ObjetoCompuesto(10);  // Se crea un objeto de la clase ObjetoCompuesto
        Base base = new Base(objeto);  // Se crea un objeto de la clase Base que usa el objeto como argumento al constructor
        ObjetoCompuesto compuesto = base.getObjetoCompuesto();  // Se obtiene una referencia al objeto ObjetoCompuesto contenido en la clase Base
        compuesto.setAtributo(20);  // Se establece el atributo en 20
        System.out.println("Atributo del objeto: " + compuesto.getAtributo());
        base.modificarAtributo(30);  // Se modifica el atributo del objeto
        System.out.println("Atributo del objeto modificado: " + compuesto.getAtributo());
        base.eliminarObjeto();  // Se elimina el objeto
    }
}
